### PR TITLE
Feat: move header to new sidebar

### LINKED ISSUE #26: frontend: Move the header to a new sidebar
## **Description**:

Move all current header elements (excluding the logo, light/dark mode toggle, wallet icon, and wallet button) to a new collapsible and responsive sidebar. This sidebar should be structured according to the specific sections listed below and use ShadCN's Sidebar component (reference: [Sidebar Component Documentation](https://ui.shadcn.com/docs/components/sidebar)). Ensure the sidebar adjusts based on authorization level (wallet connection - like the header works) and functions in both light and dark themes.

## **Requirements**:

1. Move all elements from the header (except logo, light/dark mode, wallet icon, and wallet button) to a new sidebar.
2. Make the sidebar collapsible, responsive, and display according to the user’s wallet authorization level.
3. Implement each section in the sidebar as follows:
   - **Section 1: Company Information**
     - Display the company name and icon (use an example from ShadCN).
   - **Section 2: Escrow (Collapsible) Like "Playground" in ShadCN Sidebar**
     - Implement as a collapsible section with the following items:
       - Initialize Escrow
       - Fund Escrow
       - Complete Escrow
       - Claim Escrow Earnings
       - Cancel Escrow
       - Refund Remaining Funds
       - Get Engagement
     - Use the icon `FaStackOverflow` from `react-icons/fa` for the main expandable option.
   - **Section 3: Documentation (Subtitle) Like "Projects" in ShadCN Sidebar**
     - Add the following items as external links:
       - **API Docs**: Link to [https://docs.trustlesswork.com/trustless-work](https://docs.trustlesswork.com/trustless-work)
       - **Website**: Link to [https://www.trustlesswork.com](https://www.trustlesswork.com)
     - Use the icon `FaExternalLinkAlt` from `react-icons/fa` for each link.
   - **Section 4: Profile**
     - Keep the profile section example from ShadCN.

## Validations:

1. Confirm that all header elements (except logo, light/dark mode, wallet icon, and wallet button) are successfully moved to the sidebar.
2. Verify that the sidebar is collapsible and adjusts responsively.
3. Test each section (Company Information, Escrow, Documentation, Profile) to ensure the layout and links work as expected.
4. Confirm that the sidebar respects authorization levels based on wallet connection.
5. Ensure proper styling and readability in both light and dark themes.
6. It should works on light / dark mode.
7. Must be fully responsive.

## Tests:

1. Test the sidebar’s collapsibility and responsiveness on various screen sizes.
2. Verify that the Escrow section can expand and collapse, and all items within it are accessible.
3. Confirm the external links under Documentation function and open correctly.
4. Check that the sidebar displays correctly based on wallet authorization status.
5. Test the sidebar’s appearance in both light and dark themes for consistency and readability.

## Recommendation:

For consistency, ensure that all icons and typography align with the overall UI style and maintain uniform padding and spacing for readability.

### PR DESCRIPTION
**Pull request by evgongora**
<p align="center"> CLR-S (2) </p>

# Pull Request | Trustless Work

## 1. Issue Link

- Closes #26 

---

## 2. Brief Description of the Issue

This PR implements a new collapsible and responsive sidebar that replaces most header elements while maintaining the logo, theme toggle, and wallet functionality in the header. The sidebar includes structured sections for company information, escrow operations, documentation, and profile.

## 3. Type of Change

Mark with an `x` all the checkboxes that apply (like `[x]`).

- [ ] 📝 Documentation (updates to README, docs, or comments)
- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] 👌 Enhancement (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)

---

## 4. Changes Made

1. Created new `AppSidebar` component with the following sections:

- Company Information with logo and description
- Collapsible Escrow section with all escrow-related operations
- Documentation section with external links
- Profile section showing wallet address

2. Key Components Used:

- `SidebarProvider` for state management
- `SidebarContent `for main content structure
- `SidebarFooter` for wallet information
- `SidebarGroup` for section organization
- `SidebarMenu` for navigation items
---

## 5. Evidence Before Solution

[Loom Video - Before Solution](https://www.loom.com/share/2eab9cfe09d34973858186d0a254a5d5?sid=ef978429-9245-493e-b9fb-6cd489a9c7ae)

---

## 6. Evidence After Solution

[Loom Video - After Solution](https://www.loom.com/share/b014d402273741fdbd94a895ada9a771?sid=e6498cd6-f65e-46ec-b567-bab68b69d8bf)

---

## 7. Important Notes

1. Component Structure:
Sidebar uses ShadcN UI components
Implements custom hooks for mobile detection
Uses React context for state management
2. Wallet Integration:
Sidebar displays formatted wallet address
Adapts to wallet connection state
Maintains consistent styling with header wallet display
3. Responsive Design:
Collapses to icon-only view on smaller screens
Full sidebar on desktop
Smooth transitions between states
4. Dependencies:
Requires ShadcN UI components
Integrates with existing wallet store
5. Testing Instructions:
Test sidebar collapse functionality
Verify wallet connection state handling
Check mobile responsiveness
Confirm dark/light theme compatibility

### CODE REVIEW

**Review by JoelVR17 — CHANGES_REQUESTED** [review]

File: src/components/sidebar/AppSidebar.tsx
[the remark this answers is not in the document]
Comment (evgongora) [reply src/components/sidebar/AppSidebar.tsx:1]: Understood, I will start working on it!

File: src/components/sidebar/AppSidebar.tsx [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+  isActive,
+}: {
+  address: string;
+  name: string | undefined;
+  isActive: (path: string) => boolean;
+}) => (
+  <>
+    <SidebarHeader className="border-b p-4" aria-label="Company Information">
+      <div className="flex items-center gap-2">
+        <Image src="/logo.png" alt="Trustless Work" width={24} height={24} />
```
[the remark this answers is not in the document]
Comment (evgongora) [reply src/components/sidebar/AppSidebar.tsx:75]: Got it!

**Comment by evgongora** [comment]
Made the changes! You can see it here:

https://www.loom.com/share/80eaf1a107ae4f6888c26bcc76a95ea8?sid=30953a97-275a-43f0-bb4a-1867ee18a9f6

**Comment by JoelVR17** [comment]
Merged!

Thanks for your work @evgongora!

**Pull request merged**